bolero/environment/optimum_trajectory.py

```diff
@@ -294,5 +294,116 @@ def plot(self, ax):
         if self.obstacles is not None:
             from matplotlib.patches import Circle
             for obstacle in self.obstacles:
-                ax.add_patch(Circle(obstacle, self.obstacle_dist, ec="none",
+                ax.add_patch(Circle(np.asarray(obstacle).copy(), self.obstacle_dist, ec="none",
                                     color="r"))
+
+
+class OptimumTrajectoryCurbingObstacles(OptimumTrajectory):
+    """Optimize a trajectory according to some criteria. Passing by obstacles slows down or stops the movement
```

Comment:
What does "passing obstacles" mean by the way? Does it mean you are in the obstacle or close to the obstacle?

Reply:
The effect begins when a target position is inside the obstacle_dist radius. Our current documentation of this parameter suggests that obstacles are just points (one cannot be inside the obstacle) and this is the distance to be kept. To increase clarity, I propose replacing "Passing by" with being **"in the vicinity of obstacles"** (see above).
However, to activate the feature, the user needs to change the parameter *curbing_obstacles*, which has a documentation that explains this.

Reply:
> The effect begins when a target position is inside the obstacle_dist radius. Our current documentation of this parameter suggests that obstacles are just points (one cannot be inside the obstacle) and this is the distance to be kept.

I always thought of it as circular obstacles.

> However, to activate the feature, the user needs to change the parameter _curbing_obstacles_, which has a documentation that explains this.

Then it should be clearly documented before the list of parameters in the docstring that you have to set this parameter. Otherwise an unexperienced user will wonder why the environment does not behave differently. An alternative would be to set a reasonable default value != 0.

Reply:
> I always thought of it as circular obstacles.

OK. We could change "Distance that should be kept to the obstacles" to "Radius of the obstacles that should be avoided" to reflect this. (see docstring for obstacle_dist). Then I would also change my docs from "Passing by" to "Passing through"

> a reasonable default value != 0.
Yes, the zero default was a remainder from when I implemented just an adaptation of the base class. I think 0.5 is reasonable: a single obstacle slows down, and passing trough al least two obstacles at the same time (when they overlap) will stop regardless of the inputs.